http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GKEInterpreter.scala

```diff
@@ -0,0 +1,390 @@
+            .transaction
+        } yield ()
+    }
+  }
+
+  private[monitor] def getSecrets(namespace: NamespaceName): F[List[KubernetesSecret]] =
+    config.ingressConfig.secrets.traverse { secret =>
+      for {
+        secretFiles <- secret.secretFiles
+          .traverse { secretFile =>
+            for {
+              bytes <- readFileToBytes(secretFile.path, blocker)
+            } yield (secretFile.name, bytes.toArray)
+          }
+          .map(_.toMap)
+      } yield KubernetesSecret(
+        namespace,
+        secret.name,
+        secretFiles,
+        KubernetesSecretType.Generic
+      )
+    }
+
+  private[monitor] def getGoogleNodepool(nodepool: Nodepool): com.google.container.v1.NodePool = {
+    val nodepoolBuilder = NodePool
+      .newBuilder()
+      .setConfig(
+        NodeConfig
+          .newBuilder()
+          .setMachineType(nodepool.machineType.value)
+      )
+      .setInitialNodeCount(nodepool.numNodes.amount)
+      .setName(nodepool.nodepoolName.value)
+      .setManagement(
+        NodeManagement
+          .newBuilder()
+          .setAutoUpgrade(true)
+          .setAutoRepair(true)
+      )
+
+    val builderWithAutoscaling = nodepool.autoscalingConfig.fold(nodepoolBuilder)(config =>
+      nodepool.autoscalingEnabled match {
+        case true =>
+          nodepoolBuilder.setAutoscaling(
+            NodePoolAutoscaling
+              .newBuilder()
+              .setEnabled(true)
+              .setMinNodeCount(config.autoscalingMin.amount)
+              .setMaxNodeCount(config.autoscalingMax.amount)
+          )
+        case false => nodepoolBuilder
+      }
+    )
+
+    builderWithAutoscaling.build()
+  }
+
+  def createCluster(createMessage: CreateCluster, appId: AppId)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    val createCluster =
+      for {
+        _ <- logger.info(s"beginning cluster creation for app ${appId}")
+        clusterOpt <- kubernetesClusterQuery.getMinimalClusterById(createMessage.clusterId).transaction
+        dbCluster <- F.fromOption(
+          clusterOpt,
+          KubernetesClusterNotFoundException(s"cluster with id ${createMessage.clusterId} not found in database")
+        )
+        defaultNodepool <- F.fromOption(dbCluster.nodepools.filter(_.isDefault).headOption,
+                                        DefaultNodepoolNotFoundException(dbCluster.id))
+
+        (network, subnetwork) <- vpcAlg.setUpProjectNetwork(
+          SetUpProjectNetworkParams(dbCluster.googleProject)
+        )
+
+        kubeNetwork = KubernetesNetwork(dbCluster.googleProject, network).idString
+        kubeSubNetwork = KubernetesSubNetwork(dbCluster.googleProject, dbCluster.region, subnetwork).idString
+
+        cidrBuilder = CidrBlock.newBuilder()
+        _ <- F.delay(config.clusterConfig.authorizedNetworks.foreach(cidrIP => cidrBuilder.setCidrBlock(cidrIP.value)))
```

Comment:
This line doesn't seem quite correct -- seems like it would repeatedly call `setCidrBlock` for each entry in `authorizedNetworks`? Should it be set with a comma-separated list instead?

Reply:
I tried to use a command separated list and setting the entire list. It seems that you need to set each cidrBlock separately, and a single ip is a 'cidrBlock` as far as the lib is concerned. it took a bit of iteration, and I verified that this works in the cluster via the consolle

Reply:
Hm ok.. so this doesn't result in only the _last_ one in `config.clusterConfig.authorizedNetworks` being set?